File: Top 100 Codes/02-Working with Numbers/05-Octal to Decimal conversion/OctaltoDecimalconversion.cpp
// using built-in stoi() function
#include <iostream>
#include <string>
#include <exception> // For handling exceptions
using namespace std;

int main()
{
    // Variable to store the octal number as a string
    string octalNumber;
    cout << "Enter an octal number: ";
    cin >> octalNumber;

    try {
        int base = 8; // Base 8 for octal numbers

        // Convert the octal number (string) to decimal using stoi
        int decimalValue = stoi(octalNumber, 0, base);

        // Output the converted decimal value
        cout << "Decimal value: " << decimalValue << endl;
    } catch (const invalid_argument& e) {
        // Exception thrown if the input contains invalid characters
        cout << "Error: Invalid octal number. Please enter digits from 0 to 7 only." << endl;
    } 

    return 0;
}

File: Top 100 Codes/02-Working with Numbers/05-Octal to Decimal conversion/OctaltoDecimalconversion.py
# Input: Octal number as a string
octal_number = input("Enter an octal number: ")

try:
    # Convert the octal number (base 8) to a decimal integer
    decimal_value = int(octal_number, 8)
    print(f"Decimal value: {decimal_value}")
except ValueError:
    # Handle invalid input (non-octal digits or invalid format)
    print("Error: Invalid octal number. Digits must be between 0 and 7.")
